docs/pages/references/directives/git.md

```diff
@@ -1,112 +1,382 @@
 ---
-title: Добавление кода в образ директивой git
+title: Adding source code with `git` directive
 sidebar: reference
 permalink: git.html
 folder: directive
 ---
 
 Для добавления кода в собираемый образ, предусмотрена директива `git`, с помощью которой можно добавить код из локального или удаленного репозитория (включая сабмодули) как в образ приложения так и в образ артефакта.
 
+The `git` directive in `dappfile.yml` adds source code from a Git repository to an application or artifact image.
+It supports local and remote repositories, including submodules.
+
+> Note: to work with remote git repositories, dapp requires libssh2 and/or libssl installed.
+> Refer to [Dapp Installation](installation.html#install-dependencies) for details.
+
+## Introduction
+
+### Optimizing build speed
+
 При первой сборке образа с указанием директивы `git`, в него добавляется содержимое git репозитория (стадия `g_a_archive`) согласно соответствующих инструкций. При последующих сборках образа, изменения в git репозитории добавляются отдельным docker-слоем, который содержит git-патч (git patch apply). Содержимое таких docker-слоев с патчами также кешируется, что еще более повышает скорость сборки. В случае отмены сделанных изменений в исходном коде приложения (например, через git revert), при сборке будет накладываться патч с отменой изменений, будет использоваться слой из кеша.
 
+When dapp first builds an image from a dappfile with `git` directive, it adds source code from a Git repository to the image.
+This happens on the `g_a_archive` stage.
```

Comment:
This should be link to article about build stages

Reply:
fixed